src/mongo/db/query/explain_options.cpp

```diff
@@ -80,8 +81,23 @@ StatusWith<ExplainOptions::Verbosity> ExplainOptions::parseCmdBSON(const BSONObj
                                         << kAllPlansExecutionVerbosityStr
                                         << "'}");
         }
+        verbositySpecified = true;
     }
 
+    int numFields = 0;
+    BSONObjIterator i(cmdObj);
+    while (i.moreWithEOO()) {
+        BSONElement e = i.next();
+        if (e.eoo())
+            break;
+        if (e.fieldNameStringData() == "$db")
+            continue;
+        numFields++;
+    }
+    if (numFields != (verbositySpecified ? 2 : 1)) {
```

Comment:
I think it makes more sense here to catch the invalid argument in some kind of default case while you're looping over the fields in the document. That way you're able to tell the user which argument is invalid with a more specific error message. If you do it that way you also may not need the verbositySpecified variable.

Reply:
Thank you for the pointer, I have removed the `verbositySpecified` variable.